editor/inserter/test/menu.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { registerBlockType, unregisterBlockType, getBlockTypes } from 'blocks';
+
+/**
+ * Internal dependencies
+ */
+import { InserterMenu } from '../menu';
+
+const textBlock = {
+	name: 'core/text-block',
+	title: 'Text',
+	save: noop,
+	edit: noop,
+	category: 'common',
+};
+
+const advancedTextBlock = {
+	name: 'core/advanced-text-block',
+	title: 'Advanced Text',
+	save: noop,
+	edit: noop,
+	category: 'common',
+};
+
+const moreBlock = {
+	name: 'core/more-block',
+	title: 'More',
+	save: noop,
+	edit: noop,
+	category: 'layout',
+	useOnce: 'true',
+};
+
+const youtubeBlock = {
+	name: 'core/youtube-block',
+	title: 'Youtube',
+	save: noop,
+	edit: noop,
+	category: 'embed',
+};
+
+describe( 'InserterMenu', () => {
+	afterEach( () => {
+		getBlockTypes().forEach( ( block ) => {
+			unregisterBlockType( block.name );
+		} );
+	} );
+
+	beforeEach( () => {
+		registerBlockType( textBlock.name, textBlock );
+		registerBlockType( advancedTextBlock.name, advancedTextBlock );
+		registerBlockType( moreBlock.name, moreBlock );
+		registerBlockType( youtubeBlock.name, youtubeBlock );
+	} );
+
+	it( 'Should show the recent tab by default', () => {
+		const wrapper = shallow(
+			<InserterMenu
+				instanceId={ 1 }
+				blocks={ [] }
+				recentlyUsedBlocks={ [] }
+			/>
+		);
+		const activeCategory = wrapper.find( '.editor-inserter__tab .is-active' );
+		const visibleBlocks = wrapper.find( '.editor-inserter__block' );
+		expect( activeCategory.text() ).toBe( 'Recent' );
+		expect( visibleBlocks.length ).toBe( 0 );
+	} );
+
+	it( 'Should show the recently used blocks in the recent tab', () => {
+		const wrapper = shallow(
+			<InserterMenu
+				instanceId={ 1 }
+				blocks={ [] }
+				recentlyUsedBlocks={ [ advancedTextBlock ] }
+			/>
+		);
+		const visibleBlocks = wrapper.find( '.editor-inserter__block' );
+		expect( visibleBlocks.length ).toBe( 1 );
+		expect( visibleBlocks.text() ).toBe( '<Dashicon />Advanced Text' );
```

Comment:
This fails after rebase because the Dashicon was replaced with BlockIcon in #1870. Also the way this string is formed seems potentially fragile (what if future serializers omit the space before the closing slash?). Maybe: `find`, `contains`, `childAt`, or not testing content at all.

```js
expect( visibleBlocks.childAt( 0 ).name() ).toBe( 'BlockIcon' );
expect( visibleBlocks.childAt( 1 ).text() ).toBe( 'Advanced Text' );
```